linera-chain/src/chain.rs

```diff
@@ -437,21 +437,6 @@ where
         Ok(true)
     }
 
-    /// Returns true if there are no more outgoing messages in flight up to the given
-    /// block height.
-    pub fn all_messages_delivered_up_to(&mut self, height: BlockHeight) -> bool {
-        tracing::debug!(
-            "Messages left in {:.8}'s outbox: {:?}",
-            self.chain_id(),
-            self.outbox_counters.get()
```

Comment:
Could we instead increment the counters right away for the messages to untracked chains?

Reply:
or just what you said: `tracked_outbox_counters`
perhaps we can also have: `untracked_outbox_counters` to support future changes in the tracked chains

Reply:
Actually, maybe the `outbox_counters` just need to refer only to the tracked chains. When those change, the counters have to be recomputed from the outboxes themselves anyway.

Reply:
…but then I wonder if we should move it from the (persisted) chain state view to the worker state?

Reply:
This works for me!

Reply:
(now that chain states are long lived)

Reply:
It feels error-prone to remove `outbox_counters` so far from the outboxes in the code. I'm now wondering whether we should rather add `tracked_chains` to the chain state? That way we can maintain the counters only for the tracked chains, and we can update them if the tracked chains change?

Another option is to remove the counters entirely: In practice, I'd expect the outboxes for the tracked chains to usually be empty?

Reply:
But to quickly fix the issue, I think this is by far the easiest approach.